metadata-db-app/int_test/cmr/metadata_db/int_test/concepts/variable_save_test.clj

```diff
@@ -7,20 +7,96 @@
+          ;; 3) Then wait again and save a tombstone.
+          ;; 4) Finally, wait a bit and save a new (non-tombstone) revision.
+          ;;
+          ;; All should have the same `created-at` value.
+          ;;
+          ;; Note - Originally planned to use the time-keeper functionality
+          ;; for this, but metdata-db tests don't have access to the control
+          ;; api that would allow this to work in CI.
+          {concept-id :concept-id
+           initial-revision-id :revision-id} (util/save-concept initial-var)
+          _ (Thread/sleep 100)
+          {second-revision-id :revision-id} (util/save-concept initial-var)
+          _ (Thread/sleep 100)
+          {tombstone-revision-id :revision-id} (util/delete-concept
+                                                concept-id)
+          _ (Thread/sleep 100)
+          {final-revision-id :revision-id} (util/save-concept initial-var)
+          revisions (get-revisions concept-id initial-revision-id
+                                   second-revision-id tombstone-revision-id
+                                   final-revision-id)]
+      (is (apply util/created-at-same? revisions)))))
+
+(deftest save-variable-with-conflicting-native-id
+  (testing "Save a variable"
+    (let [var1 (util/variable-concept "PROV1" 1)
+          {:keys [status revision-id concept-id]} (util/save-concept var1)]
+      (is (= status 201))
+      (is (= 1 revision-id))
+      (testing "and another with all the same data"
+        (let [var2 var1
+              {:keys [status revision-id]} (util/save-concept var1)]
+          (is (= 201 status))
+          (is (= 2 revision-id))))
+      (testing "and another with same data but different provider"
+        (let [var3 (util/variable-concept "PROV2" 1)
+              {:keys [status revision-id]} (util/save-concept var3)]
+          (is (= status 201))
+          (is (= 1 revision-id))))
+      (testing "and another the same data but with a different native-id"
+        (let [var4 (assoc var1 :native-id "var-native-different")
+              failed-concept (util/save-concept var4)
+              failed-revision-id 2
+              failed-concept-id "V1200000002-PROV1"]
+          (is (= 409 (:status failed-concept)))
+          (is (= [(format (str "Revision [%s] of concept [%s] has the same "
+                               "provider [%s] and variable name [%s] but "
+                               "different native id [%s] than the new concept "
+                               "[%s] with revision [%s] and native id [%s].")
+                          failed-revision-id
+                          concept-id
+                          (:provider-id var1)
+                          (get-in var1 [:extra-fields :variable-name])
+                          (:native-id var1)
+                          failed-concept-id
+                          revision-id
+                          (:native-id var4))]
+                  (:errors failed-concept))))))))
```

Comment:
We should add a check to verify that the new revision-id that CMR tries to create has indeed been rolled back, i.e. does not exist in metadata-db any more.